### PR TITLE
removed any types where possible

### PR DESCRIPTION
**Pull request by techcontributor**
- Removed any types
- Added types where needed
- Minor code optimisation

### CODE REVIEW

File: lib/types/index.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ export type SearchResultItem = {
   title: string
   url: string
   content: string
+  author?: string
```
Comment (miurla) [inline lib/types/index.ts]: The Tavily API response does not include the `author`.

https://docs.tavily.com/docs/tavily-api/rest_api#example-response

**Review by miurla — COMMENTED** [review]
Everything except the comments is OK

**Comment by techcontributor** [comment]
> Everything except the comments is OK

Fine, let me change and update the pr

**Comment by techcontributor** [comment]
@miurla now please check

**Review by miurla — APPROVED** [review]
Thank you for your contribution!

**Pull request merged**